### PR TITLE
[Shell] Fix Register route and Navigate to it

### LINKED ISSUE #5466: [Shell] Cannot navigate to a route
### Description

No matter what I try to navigate to by URI, I get this error:

```
System.NotImplementedException: The method or operation is not implemented.
  at Xamarin.Forms.Shell.GoToAsync (Xamarin.Forms.ShellNavigationState state, System.Boolean animate) [0x000d8] in D:\a\1\s\Xamarin.Forms.Core\Shell\Shell.cs:382 
  at Demos.ViewModels.DemoViewModel.HandleVisualNav (System.String page) [0x00035] in /work/TheLittleThingsPlayground/TheLittleThingsPlayground/ViewModels/DemoViewModel.cs:35 
```

In AppShell.xaml.cs I register the route like this:

```
Routing.RegisterRoute("buttons", typeof(ButtonsPage));
```

And then in the VM call the `GoToAsync` (page is "buttons"):

```
await Shell.CurrentShell.GoToAsync($"{page}", true);
```

I've also tried:

```
await Shell.CurrentShell.GoToAsync($"app://root/ShellItem14/three_six/{page}", true);
await Shell.CurrentShell.GoToAsync($"app:///ShellItem14/three_six/{page}", true);
await Shell.CurrentShell.GoToAsync($"/{page}", true);
```

Thinking I could use the relative navigation that @paymicro added in #4977 . 

### Basic Information

- Version with issue: 3.6.0.220655

### Reproduction

https://github.com/davidortinau/TheLittleThingsPlayground/blob/3.6.0/TheLittleThingsPlayground/ViewModels/DemoViewModel.cs#L35

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

Small fixes to shell, when trying to fix routing and navigate to that page. 

- Allow to set the SafeAreaInsets on Shell itself so it's used for all pages
- Fix routing
- Fix NRE when page is Null because it was set later

### Issues Resolved ### 

- fixes #5466

### API Changes ###

 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Controls/XamStore/StoreShell.xaml.cs [hunk 1]
```diff
@@ -12,12 +12,13 @@ namespace Xamarin.Forms.Controls.XamStore
 	[XamlCompilation(XamlCompilationOptions.Compile)]
 	public partial class StoreShell : Shell
 	{
-		public StoreShell ()
-		{
-			InitializeComponent ();
+			public StoreShell ()
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Controls/XamStore/StoreShell.xaml.cs]: what's going on with the indentation ?

**Comment by StephaneDelcroix** [comment]
@rmarinho please provide an accurate PR title and description. I'm fine with the one line change related to Insets, I don't know what the other changes are.

File: Xamarin.Forms.Core/Shell/Shell.cs [hunk 2]
```diff
@@ -335,9 +335,9 @@ Uri GetAbsoluteUri(Uri relativeUri)
 			var query = parseUri["q"].Value;
 			var fragment = parseUri["f"].Value;
 
-			Element item = CurrentItem;
+			Element item = ignoreCurrentItem ? CurrentItem.Parent : CurrentItem;
```
Comment (paymicro) [inline Xamarin.Forms.Core/Shell/Shell.cs]: I think it is not quite right. If the idea is to ignore the current item, then it parent will not always be its root.
The best solution is to remove all items from the list between the first and the last. 
`if (list.Count > 2) list.RemoveRange(1, list.Count - 2);`

**Comment by rmarinho** [comment]
> @rmarinho please provide an accurate PR title and description. I'm fine with the one line change related to Insets, I don't know what the other changes are.

sorry, I started the call and forget to continue the update.. Added more info @StephaneDelcroix

File: Xamarin.Forms.Controls/XamStore/Styles/HomeView.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ListView {
+    background-color: #11313F;
+}
+
+.AddButton {
+    background-color: transparent;
+    width: 60;
+    height: 60;
+    border: 0;
+}
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Controls/XamStore/Styles/HomeView.css]: ❤️

**Pull request merged**